src/ucp/rma/basic_rma.c

```diff
@@ -304,9 +304,62 @@ ucs_status_t ucp_get_nbi(ucp_ep_h ep, void *buffer, size_t length,
     return UCS_INPROGRESS;
 }
 
+static ucs_status_t ucp_progress_fence(uct_pending_req_t *self)
+{
+    ucp_request_t *req = ucs_container_of(self, ucp_request_t, fence.uct);
+    ucp_ep_t *ep = req->fence.ep;
+    ucs_status_t status;
+
+    status = uct_ep_fence(ep->uct_eps[req->fence.lane], 0);
+    if (status == UCS_ERR_NO_RESOURCE)
+        return UCS_INPROGRESS;
+    else if (status == UCS_OK) {
+        ucp_request_put(req);
+        return UCS_OK;
+    }
+    else {
+        return status;
+    }
+}
+
 ucs_status_t ucp_worker_fence(ucp_worker_h worker)
 {
-    return UCS_ERR_UNSUPPORTED;
+    ucs_status_t status;
+    ucp_ep_h ep;
+    uct_ep_h uct_ep;
+    ucp_lane_index_t lane;
+
+    kh_foreach_value(&worker->ep_hash, ep,
```

Comment:
@yosefe pending queue is per ep, so if current uct_iface_fence returns NO_RESOURCE, where should we add the request? To the first ep in current iface?